app/controllers/network_router_controller.rb

```diff
@@ -167,6 +167,8 @@ def delete_network_routers
       router = NetworkRouter.find_by(:id => s)
       if router.nil?
         add_flash(_("Router no longer exists."), :error)
+      elsif !router.supports?(:delete)
+        add_flash(_("%{reason}") % {:reason => router.unsupported_reason(:delete)}, :error)
```

Comment:
I'd say, as long as `router.unsupported_reason(:delete)` returns a string, you can just as well use:
```ruby
add_flash(_(router.unsupported_reason(:delete)), :error)
```